src/arm/radxa_rock3c.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Author: Ken <ken@radxa.com>
+ * Copyright (c) Radxa Limited.
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <mraa/common.h>
+#include <stdarg.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include "arm/radxa_rock3c.h"
+#include "common.h"
+
+#define DT_BASE "/proc/device-tree"
+
+#define PLATFORM_NAME_RADXA_ROCK_3_MODEL_C "Radxa ROCK3 Model C"
+
+#define MAX_SIZE 64
+
+const char* radxa_rock3c_serialdev[MRAA_RADXA_ROCK3C_UART_COUNT] = { "/dev/ttyS3", "/dev/ttyS4", "/dev/ttyS5", "/dev/ttyS9" };
+
+void
+mraa_radxa_rock3c_pininfo(mraa_board_t* board, int index, int sysfs_pin, mraa_pincapabilities_t pincapabilities_t, char* fmt, ...)
+{
+    va_list arg_ptr;
+
+    if (index > board->phy_pin_count)
+        return;
+
+    mraa_pininfo_t* pininfo = &board->pins[index];
+    va_start(arg_ptr, fmt);
+    vsnprintf(pininfo->name, MRAA_PIN_NAME_SIZE, fmt, arg_ptr);
+
+    if( pincapabilities_t.gpio == 1 ) {
+        va_arg(arg_ptr, int);
+        pininfo->gpio.gpio_chip = va_arg(arg_ptr, int);
+        pininfo->gpio.gpio_line = va_arg(arg_ptr, int);
+    }
+
+    pininfo->capabilities = pincapabilities_t;
+
+    va_end(arg_ptr);
+    pininfo->gpio.pinmap = sysfs_pin;
+    pininfo->gpio.mux_total = 0;
+}
+
+mraa_board_t*
+mraa_radxa_rock3c()
+{
+    mraa_board_t* b = (mraa_board_t*) calloc(1, sizeof(mraa_board_t));
+    if (b== NULL) {
+        return NULL;
+    }
+
+    b->adv_func = (mraa_adv_func_t*) calloc(1, sizeof(mraa_adv_func_t));
+    if (b->adv_func == NULL) {
+        free(b);
+        return NULL;
+    }
+
+    // pin mux for buses are setup by default by kernel so tell mraa to ignore them
+    b->no_bus_mux = 1;
+    b->phy_pin_count = MRAA_RADXA_ROCK3C_PIN_COUNT + 1;
+
+    if (mraa_file_exist(DT_BASE "/model")) {
+        // We are on a modern kernel, great!!!!
+        if (mraa_file_contains(DT_BASE "/model", PLATFORM_NAME_RADXA_ROCK_3_MODEL_C)  ) {
+            b->platform_name = PLATFORM_NAME_RADXA_ROCK_3_MODEL_C;
+            b->uart_dev[0].device_path = (char*) radxa_rock3c_serialdev[0];
+            b->uart_dev[1].device_path = (char*) radxa_rock3c_serialdev[1];
+            b->uart_dev[2].device_path = (char*) radxa_rock3c_serialdev[2];
+            b->uart_dev[3].device_path = (char*) radxa_rock3c_serialdev[3];
+        }
+    }
+
+    // UART
+    b->uart_dev_count = MRAA_RADXA_ROCK3C_UART_COUNT;
+    b->def_uart_dev = 0;
+    b->uart_dev[0].index = 3;
+    b->uart_dev[1].index = 4;
+    b->uart_dev[2].index = 5;
+    b->uart_dev[3].index = 9;
```

Comment:
```suggestion
    b->platform_name = PLATFORM_NAME_RADXA_ROCK_3_MODEL_C;

    // pin mux for buses are setup by default by kernel so tell mraa to ignore them
    b->no_bus_mux = 1;
    b->phy_pin_count = MRAA_RADXA_ROCK3C_PIN_COUNT + 1;

    // UART
    b->uart_dev_count = MRAA_RADXA_ROCK3C_UART_COUNT;
    b->def_uart_dev = 0;
    b->uart_dev[0].index = 3;
    b->uart_dev[1].index = 4;
    b->uart_dev[2].index = 5;
    b->uart_dev[3].index = 9;
    b->uart_dev[0].device_path = (char*) radxa_rock3c_serialdev[0];
    b->uart_dev[1].device_path = (char*) radxa_rock3c_serialdev[1];
    b->uart_dev[2].device_path = (char*) radxa_rock3c_serialdev[2];
    b->uart_dev[3].device_path = (char*) radxa_rock3c_serialdev[3];
```
这些设备树的判断在3C上应该是不必要的。`arm.c`里面已经判断过了。